common/src/test/java/org/conscrypt/TrustManagerImplTest.java

```diff
@@ -135,76 +134,32 @@ public void testHttpsEndpointIdentification() throws Exception {
         String goodHostname = TestKeyStore.CERT_HOSTNAME;
         String badHostname = "definitelywrong.nopenopenope";
 
-        // The default hostname verifier on OpenJDK rejects all hostnames, so use our own
-        javax.net.ssl.HostnameVerifier oldDefault = HttpsURLConnection.getDefaultHostnameVerifier();
         try {
-            HttpsURLConnection.setDefaultHostnameVerifier(new TestHostnameVerifier());
-
             SSLParameters params = new SSLParameters();
 
             // Without endpoint identification this should pass despite the mismatched hostname
             params.setEndpointIdentificationAlgorithm(null);
 
             List<X509Certificate> certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+                    new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
             assertEquals(Arrays.asList(chain), certs);
 
             // Turn on endpoint identification
             params.setEndpointIdentificationAlgorithm("HTTPS");
 
-            try {
-                tmi.getTrustedChainForServer(chain, "RSA",
-                    new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+            try { // this should fail
+                certs = tmi.getTrustedChainForServer(chain, "RSA",
+                        new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+                assertEquals(Arrays.asList(chain), certs);
                 fail();
-            } catch (CertificateException expected) {
-            }
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            // Override the global default hostname verifier with a Conscrypt-specific one that
-            // always passes.  Both scenarios should pass.
-            Conscrypt.setDefaultHostnameVerifier(new ConscryptHostnameVerifier() {
-                @Override public boolean verify(String s, SSLSession sslSession) { return true; }
-            });
 
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            // Now set an instance-specific verifier on the trust manager.  The bad hostname should
-            // fail again.
-            Conscrypt.setHostnameVerifier(tmi, new TestHostnameVerifier());
```

Comment:
And you'll probably need to change this to something like `Conscrypt.wrapHostnameVerifier(new org.conscrypt.javax.net.ssl.TestHostnameVerifier())`